tests/integration/distribution.go

```diff
@@ -103,13 +112,29 @@ func (s *CCVTestSuite) TestRewardsDistribution() {
 	// Set the consumer reward denom. This would be done by a governance proposal in prod
 	s.providerApp.GetProviderKeeper().SetConsumerRewardDenom(s.providerCtx(), rewardCoins[ibcCoinIndex].Denom)
 
-	s.providerChain.NextBlock()
+	// Refill the consumer fee pool
+	err = consumerBankKeeper.SendCoinsFromAccountToModule(s.consumerCtx(), s.consumerChain.SenderAccount.GetAddress(), authtypes.FeeCollectorName, fees)
+	s.Require().NoError(err)
+
+	// pass two blocks
+	s.consumerChain.NextBlock()
+	s.consumerChain.NextBlock()
+
+	// transfer rewards from consumer to provider
+	relayAllCommittedPackets(s, s.consumerChain, s.transferPath, transfertypes.PortID, s.transferPath.EndpointA.ChannelID, 1)
+
+	// check that the consumer rewards allocation are empty since relayAllCommittedPackets call BeginBlock
```

Comment:
Not sure I fully understood this comment. It seems to me that the consumer rewards allocation should be empty because we did not have a `OnRecvPacket` yet. Why does `BeginBlock` come into play?

Reply:
`relayAllCommittedPackets` relays the IBC packets and hence executes successively the IBC transfer middleware and BeginBlock. Therefore, the provider module's `AllocateTokens` logic is executed and the consumer allocations are emptied.